### PR TITLE
Added Grommet Tools header example in storybook

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Added the header from the grommet tools site as a storybook example

#### What testing has been done on this PR?
Storybook

#### What are the relevant issues?
Issue #4385 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/Header/stories/Responsive.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        margin={{ left: 'medium', vertical: 'xsmall' }}
+        href="https://tools.grommet.io/"
+        icon={<GrommetIcon color="brand" />}
+        label="Grommet Tools"
+      />
+      <ResponsiveContext.Consumer>
+        {size =>
+          size === 'small' ? (
+            <Box alignSelf="end" margin="large">
+              <Menu
```
Comment (ShimiSun) [inline src/js/components/Header/stories/Responsive.js:21]: Menu isn't aligned with the main link
image

File: src/js/components/Header/stories/Responsive.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      <Anchor
+        margin={{ left: 'medium', vertical: 'xsmall' }}
+        href="https://tools.grommet.io/"
+        icon={<GrommetIcon color="brand" />}
+        label="Grommet Tools"
+      />
+      <ResponsiveContext.Consumer>
+        {size =>
+          size === 'small' ? (
+            <Box alignSelf="end" margin="large">
```
Comment (ShimiSun) [inline src/js/components/Header/stories/Responsive.js:20]: Not sure why this margin is needed. we can remove it and simplify it, same for line #44.

File: src/js/components/Header/stories/Responsive.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            <Box justify="end">
+              <Menu
+                a11yTitle="Navigation Menu"
+                dropProps={{ align: { top: 'bottom', right: 'right' } }}
+                icon={<MenuIcon color="brand" />}
+                items={[
+                  {
+                    label: <Box pad="small">Grommet Tools </Box>,
+                    href: 'https://tools.grommet.io/',
+                  },
```
Comment (ShimiSun) [inline src/js/components/Header/stories/Responsive.js:28]: Not sure why we need this entry? It doesn't exist on the large screen option where you have only two links.

Comment (jcfilben) [reply src/js/components/Header/stories/Responsive.js]: Removed in latest commit

**Pull request merged**